core/src/main/java/io/questdb/cairo/mv/MatViewRefreshJob.java

```diff
@@ -462,24 +465,54 @@ private boolean insertAsSelect(
                     assert cursorTimestampIndex > -1;
 
                     long commitTarget = batchSize;
-                    rowCount = 0;
+                    long rowCount = 0;
+
 
                     intervalIterator.toTop(intervalStep);
+                    long replacementTimestampLo = Long.MIN_VALUE;
+                    long replacementTimestampHi = Long.MIN_VALUE;
+
                     while (intervalIterator.next()) {
                         refreshExecutionContext.setRange(intervalIterator.getTimestampLo(), intervalIterator.getTimestampHi());
+                        if (replacementTimestampHi != intervalIterator.getTimestampLo()) {
+                            if (replacementTimestampHi > replacementTimestampLo) {
+                                // Gap in the refresh intervals, commit the previous batch
+                                // so that the replacement interval does not span across the gap.
+                                refreshFinishTimestamp = microsecondClock.getTicks();
+                                walWriter.commitMatView(commitBaseRefreshTxn, refreshFinishTimestamp, replacementTimestampLo, replacementTimestampHi);
```

Comment:
We should only commit the last refresh base txn (`commitBaseRefreshTxn`) in the very last commit here like it was prior to this patch. Otherwise, if we crash/are force-stopped/get an OOM in the middle of a refresh, the mat view state could be left with a bumped base txn while the actual refresh didn't happen fully.

Reply:
Ok, I think I made it the same as it was in [e5a0e27](https://github.com/questdb/questdb/pull/5679/commits/e5a0e27f64130079e142e74961624ef35a9d0240)

Please have another look